src/runtime-rs/crates/hypervisor/src/device/mod.rs

```diff
@@ -49,7 +49,7 @@ impl fmt::Display for DeviceType {
 #[async_trait]
 pub trait Device: std::fmt::Debug + Send + Sync {
     // attach is to plug device into VM
-    async fn attach(&mut self, h: &dyn hypervisor) -> Result<()>;
+    async fn attach(&mut self, h: &dyn hypervisor) -> Result<DeviceType>;
```

Comment:
I'm struggling a bit trying to understand the motivation of this change.  The function already takes `&mut self`, wouldn't it be enough for implementations just to mutate `self`?  Or is it envisioned that an implementation might return a different device than `self`?

Reply:
fixed